jetty-util/src/main/java/org/eclipse/jetty/util/OpaqueIndex.java

```diff
@@ -0,0 +1,62 @@
+//
+// ========================================================================
+// Copyright (c) 1995-2020 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under
+// the terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0
+//
+// This Source Code may also be made available under the following
+// Secondary Licenses when the conditions for such availability set
+// forth in the Eclipse Public License, v. 2.0 are satisfied:
+// the Apache License v2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.util;
+
+import java.util.LinkedHashSet;
+import java.util.Set;
+
+public interface OpaqueIndex
+{
+
+    boolean contains(String key);
+
+    class Builder
```

Comment:
I'm not yet decided about the `Builder`s API, I'm considering merging them all into a single one. But the `OpaqueIndex` is needed because it exposes these two methods:

```java
    boolean contains(String key);
    boolean contains(String key, int offset, int length);
```
and it could also expose some forms of `boolean containsBest(...)` in the future. But I'm not thrilled by this interface's name either.